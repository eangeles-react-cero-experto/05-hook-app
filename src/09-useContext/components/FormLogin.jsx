import { useForm } from "../hooks/index.js";
import { useContext } from "react";
import { UserContext } from "../context/UserContext.jsx";

export const FormLogin = () => {
  const { setUser } = useContext(UserContext);
  const { formRef, formState, onSubmit } = useForm({
    username: "",
    email: "",
  });

  onSubmit(() => {
    setUser(formState);
  });

  return (
    <form ref={formRef}>
      <input
        type="text"
        name="username"
        placeholder="username"
        className="form-control"
        defaultValue={formState.username}
      />
      <br />
      <input
        type="email"
        name="email"
        placeholder="email"
        className="form-control"
        defaultValue={formState.email}
      />
      <br />
      <br />
      <button type="submit" className="btn btn-primary">
        &nbsp;&nbsp;Login&nbsp;&nbsp;
      </button>
    </form>
  );
};
