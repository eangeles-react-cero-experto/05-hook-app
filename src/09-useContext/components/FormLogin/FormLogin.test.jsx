import { fireEvent, render, screen } from '@testing-library/react'
import { UserContext } from '../../context/UserContext.jsx'
import { FormLogin } from './FormLogin.jsx'
// import { useForm } from "../../hooks/index.js";

// jest.mock("../../hooks/index.js");

describe("Test FormLogin", () => {
  const user = {
    username: "John Due",
    email: "john@email.com",
  };

  // useForm.mockReturnValue({
  //   formRef: null,
  //   formState: user,
  //   onSubmit: jest.fn(),
  // });

  beforeEach(() => jest.clearAllMocks());

  it("should match with snapshot", () => {
    const { container } = render(
      <UserContext.Provider value={{ user }}>
        <FormLogin />
      </UserContext.Provider>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should call setUser when button is pressed", () => {
    const mockSetUser = jest.fn();

    render(
      <UserContext.Provider value={{ user, setUser: mockSetUser }}>
        <FormLogin></FormLogin>
      </UserContext.Provider>,
    );

    const loginButtonElement = screen.getByText("Login");
    fireEvent.click(loginButtonElement);

    expect(mockSetUser).toHaveBeenCalledTimes(1);
    // expect(mockSetUser).toHaveBeenCalledWith(user);
  });
});
