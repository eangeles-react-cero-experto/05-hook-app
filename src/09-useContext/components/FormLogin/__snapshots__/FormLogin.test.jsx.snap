// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FormLogin should match with snapshot 1`] = `
<div>
  <form>
    <input
      class="form-control"
      name="username"
      placeholder="username"
      type="text"
      value=""
    />
    <br />
    <input
      class="form-control"
      name="email"
      placeholder="email"
      type="email"
      value=""
    />
    <br />
    <br />
    <button
      class="btn btn-primary"
      type="submit"
    >
        Login  
    </button>
  </form>
</div>
`;
